src/hooks/useSlidingSyncRoomSearch.ts

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useCallback, useState } from "react";
+import { Room } from "matrix-js-sdk/src/models/room";
+
+import { MatrixClientPeg } from "../MatrixClientPeg";
+import { useLatestResult } from "./useLatestResult";
+import { SlidingSyncManager } from "../SlidingSyncManager";
+
+export interface ISlidingSyncRoomSearchOpts {
```

Comment:
Our new [code style](https://github.com/vector-im/element-meta/pull/216/files) :magic_wand: says interfaces without `I`

```suggestion
export interface SlidingSyncRoomSearchOpts {
```